docs/concepts/callbacks.md

```diff
@@ -1,2 +1,226 @@
+```
+
+## EarlyStopping
+
+The `EarlyStopping` callback runs at the end of each training epoch and decides,
+based on user-provided criteria, if it should stop training and save the current model.
+
+It takes `monitor` and `mode` parameters that work the same way they do for `BestModelCheckpoint` above.
+However, instead of using this information to identify the best-performing model out of all epochs, it uses user-specified parameters at the end of each epoch to determine if it should terminate fine-tuning and save the current model.
+
+Below is the output of a Finetuner run with early stopping, compared to one with the same data parameters but without early stopping.
+Note that the early stopping run terminated after ten epochs, while the other run continued for the whole twenty epochs. The loss of the final model is, in the two cases, nearly identical.
+The early stopping run has a loss of 0.284, while taking the whole 20 epochs produced a model with a loss of 0.272.
+Early stopping halved the runtime at a cost to performance of less than five percent.
+
+```bash
+  Training [10/20] ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54/54 0:00:00 0:00:14 • loss: 0.284
+[11:19:28] INFO     Done ✨                                                                              __main__.py:194
+           DEBUG    Finetuning took 0 days, 0 hours 2 minutes and 30 seconds                             __main__.py:196
+           INFO     Building the artifact ...                                                            __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                                       __main__.py:231
+```
+
+```bash
+  Training [20/20] ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54/54 0:00:00 0:00:14 • loss: 0.272
+[10:37:33] INFO     Done ✨                                                                              __main__.py:194
+           DEBUG    Finetuning took 0 days, 0 hours 4 minutes and 54 seconds                             __main__.py:196
+           INFO     Building the artifact ...                                                            __main__.py:207
+           INFO     Pushing artifact to Hubble ...                                                       __main__.py:231
+```
+
+The user-specified parameters are:
+
+- `min_delta`: The minimum amount of improvement that a model must have over the previous best model in order for fine-tuning to continue. By default, this is zero, meaning that the training will not stop early unless the performance starts to decrease.
+- `patience`: The number of consecutive rounds without improvement before the training is stopped, set to two by default.
+- `baseline`: an optional parameter that is used to compare the model's score against instead of the best previous model when checking for improvement. If specified, the improvement every epoch is measured by comparison to this value instead of the best-performing model so far.
```

Comment:
```suggestion
- `baseline`: an optional parameter against which the model's score is compared when checking for improvement, instead of the best previous model. If specified, the improvement every epoch is measured by comparison to this value instead of the best-performing model so far.
```